src/main/scala/AddressLookup.scala

```diff
@@ -46,19 +46,18 @@ object AddressLookup {
               .toList
   }
 
-  def listAllProductsAvailableToDownload(requestedEpoch: String): Seq[OSGBProduct] =
+  def listAllProductsAvailableToDownload(requestedEpoch: Option[String]): Seq[OSGBProduct] =
     AddressLookup.productTypes.flatMap { p =>
       if (requestedEpoch.isEmpty)
         sardineWrapper.exploreRemoteTree.findLatestFor(p)
       else
-        sardineWrapper.exploreRemoteTree.findAvailableFor(p, requestedEpoch.toInt)
+        sardineWrapper.exploreRemoteTree.findAvailableFor(p, requestedEpoch.map(_.toInt).get)
     }
 
-  def listFiles(requestedEpoch: String): AddressLookupFileListResponse = {
+  def listFiles(requestedEpoch: Option[String], forceDownload: Boolean): AddressLookupFileListResponse = {
     val products = listAllProductsAvailableToDownload(requestedEpoch)
-    val epoch = if (requestedEpoch.nonEmpty) requestedEpoch else products.head.epoch.toString
-
-    val filesAlreadyDownloaded = AddressLookup.listAllDoneFiles(epoch)
+    val epoch = requestedEpoch.fold(products.head.epoch.toString)(identity)
+    val filesAlreadyDownloaded = if (forceDownload) Seq() else AddressLookup.listAllDoneFiles(epoch)
```

Comment:
If we don't delete at least the done files we don't easily know where things go wrong if the downloads fail. I think we should delete at least the done files. And if we need to delete those then we can just delete everything anyway.